src/state/validation.py

```diff
@@ -60,13 +61,15 @@ def wrapper(instance: C, *args: typing.Any) -> None:
             except (
                 state.config.InvalidCharmConfigError,
                 state.tls.TlsIntegrationMissingError,
+                state.config.GatewayClassUnavailableError,
+                InsufficientPermissionError,
```

Comment:
Might be interesting now to have a parent exception class to simplify this

Reply:
Yes, I added this in a future PR.